sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubClientBuilderTest.java

```diff
@@ -125,6 +162,115 @@ public void testProxyOptionsConfiguration(String proxyConfiguration, boolean exp
         Assertions.assertEquals(expectedClientCreation, clientCreated);
     }
 
+    @Test
+    public void sendAndReceiveEventByAzureNameKeyCredential() {
+        ConnectionStringProperties properties = getConnectionStringProperties();
+        String fullyQualifiedNamespace = getFullyQualifiedDomainName();
+        String sharedAccessKeyName = properties.getSharedAccessKeyName();;
+        String sharedAccessKey = properties.getSharedAccessKey();
+        String eventHubName = getEventHubName();
+
+        final EventData testData = new EventData(TEST_CONTENTS.getBytes(UTF_8));
+
+        EventHubProducerClient asyncProducerClient = new EventHubClientBuilder()
+            .credential(fullyQualifiedNamespace, eventHubName,
+                new AzureNamedKeyCredential(sharedAccessKeyName, sharedAccessKey))
+            .buildAsyncProducerClient();
+        asyncProducerClient.send(testData);
```

Comment:
This will never be sent because it needs to be subscribed to.

```java
try {
    StepVerifier.create(asyncProducerClient.send(testData))
        .verifyComplete();
} finally {
    // Please close client regardless of success or failure because we need to clean up resources.
    asyncProducerClient.close();
}
```

You don't have to create the asyncConsumerClient and receive below. If the send does not complete in the lines above, the credential is invalid.

Reply:
According to your comment in the new revision amendments have been made.